cmd/admin-bucket-replication-set.go

```
@@ -0,0 +1,202 @@
+	if argCount < 2 {
+		return sourceBucket, replTarget, fmt.Errorf("Missing replication target configuration")
+	}
+	_, sourceBucket = url2Alias(args[0])
+	replicaURL := args[1]
+	api := cli.String("api")
+	if api != "" && !isValidAPI(api) { // Empty value set to default "S3v4".
+		fatalIf(errInvalidArgument().Trace(api),
+			"Unrecognized API signature. Valid options are `[S3v4, S3v2]`.")
+	}
+	path := cli.String("path-style")
+	if !isValidPath(path) {
+		fatalIf(errInvalidArgument().Trace(path),
+			"Unrecognized bucket path style. Valid options are `[on,off, auto]`.")
+	}
+	u, cerr := url.Parse(replicaURL)
+	if cerr != nil {
+		fatalIf(probe.NewError(cerr), "Malformed replication target URL")
+	}
+	isSSL := u.Scheme == "https"
+	accessKey := u.User.Username()
+	secretKey, _ := u.User.Password()
+	replicaBucket := strings.TrimPrefix(u.Path, slashSeperator)
+	console.SetColor(cred, color.New(color.FgYellow, color.Italic))
+	creds := &auth.Credentials{AccessKey: accessKey, SecretKey: secretKey}
+	replTarget = &madmin.BucketReplicationTarget{TargetBucket: replicaBucket, IsSSL: isSSL, Credentials: creds, Endpoint: u.Host, Path: path, API: api}
+	return sourceBucket, replTarget, nil
+}
+
+// mainAdminBucketReplicationSet is the handle for "mc admin bucket replication set" command.
+func mainAdminBucketReplicationSet(ctx *cli.Context) error {
+	checkAdminBucketReplicationSetSyntax(ctx)
+
+	console.SetColor("ReplicationMessage", color.New(color.FgGreen))
+
+	// Get the alias parameter from cli
+	args := ctx.Args()
+	aliasedURL := args.Get(0)
+	// Create a new MinIO Admin Client
+	client, cerr := newAdminClient(aliasedURL)
+	fatalIf(cerr, "Unable to initialize admin connection.")
+
+	sourceBucket, replTarget, perr := fetchReplicationTarget(ctx)
+	fatalIf(probe.NewError(perr), "Unable to parse input arguments.")
+	fatalIf(probe.NewError(client.SetBucketReplicationTarget(globalContext, sourceBucket, replTarget)).Trace(args...), "Cannot add new replication target")
```

Just a q: why is replication not supported in fs mode? It should be noted in the [server guide](https://github.com/minio/minio/blob/c1e3495f82a664a46a8835741b990a2b1d945cc0/docs/bucket/replication/README.md).

---

replication needs versioning enabled on both source and destination - FS mode doesn't support versioning.

---

Ah ok, i was able to enable versioning via mc for fs mode - the server should return an error for that.

---

yes that should be the case @donatello